File: php/src/Services/BasicInfoService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;
use CCPayment\APIError;
use GuzzleHttp\Exception\GuzzleException;

class BasicInfoService
{
    private Client $client;
    public function __construct(Client $client) { $this->client = $client; }
    public function getCoinList(): array { return $this->client->post('/getCoinList', []); }
    public function getFiatList(): array { return $this->client->post('/getFiatList', []); }
    public function getCoin(int $coinId): array { return $this->client->post('/getCoin', ['coinId' => $coinId]); }
    public function getCoinUSDTPrice(array $coinIds): array { return $this->client->post('/getCoinUSDTPrice', ['coinIds' => $coinIds]); }
    public function getChainList(array $data = []): array { return $this->client->post('/getChainList', $data); }
    public function getAllChainList(array $data = []): array { return $this->client->post('/all/chain', $data); }
    public function getMainCoinList(string $appId): array { return $this->client->post('/getMainCoinList', ['appId' => $appId]); }
}

File: ruby/lib/ccpayment/services/basic_info_service.rb
module CCPayment
  module Services
    class BasicInfoService
      def initialize(client); @client = client; end
      def get_coin_list; @client.post('/getCoinList', {}); end
      def get_fiat_list; @client.post('/getFiatList', {}); end
      def get_coin(coin_id); @client.post('/getCoin', { coinId: coin_id }); end
      def get_coin_usdt_price(coin_ids); @client.post('/getCoinUSDTPrice', { coinIds: coin_ids }); end
      def get_chain_list(data = {}); @client.post('/getChainList', data); end
      def get_all_chain_list(data = {}); @client.post('/all/chain', data); end
      def get_main_coin_list(app_id); @client.post('/getMainCoinList', { appId: app_id }); end
    end
  end
end
